src/buildApplication/buildWorkerFile.ts

```diff
@@ -4,12 +4,28 @@ import { build } from 'esbuild';
 import { tmpdir } from 'os';
 import { cliLog } from '../cli';
 import { NextJsConfigs } from './nextJsConfigs';
-import { MiddlewareManifestData } from './middlewareManifest';
 import { generateGlobalJs } from './generateGlobalJs';
+import { ProcessedVercelOutput } from './processVercelOutput';
 
+/**
+ * Construct a record for the build output map.
+ *
+ * @param item The build output item to construct a record for.
+ * @returns Record for the build output map.
+ */
+function constructBuildOutputRecord(item: BuildOutputItem) {
```

Comment:
If you're okay with me putting type assertions for everything then sure, although the type safety is kind of not really there if we're having to put `x as ...` for each property value. The problem is all the property values need to be strings to be able to be consumed as valid JS when we inject it.

Reply:
ah no sorry you're right... you are producing strings 🤔 (sorry thought you were returning the objects)

You could just return the objects and have the types be correct, then do a stringify where you use the results or something like that 🤔 

Well not a big deal, I'm totally fine with this 🙂

Reply:
Well both the `type` value and `entrypoint` value can't strictly match up with the actual type that it would be, so returning the object with the correct types would still need to heavily use assertions, and then you could return the object without stringifying it and stringify it later, but at that point, I feel like it's more beneficial to just remove the function and do it all in the same place like before.

I just wanted to separate the creating the stringified object from the actual consumption so it was a bit more readable - sadly I think this is the only thing that won't get real type safety 😅